php/Block.php

```diff
@@ -48,81 +41,32 @@ public function __construct( $plugin ) {
 	 * Inits the class.
 	 */
 	public function init() {
-		add_action( 'init', [ $this, 'register_block' ] );
-	}
-
-	/**
-	 * Registers the block as a dynamic block, with a render_callback.
-	 */
-	public function register_block() {
-		if ( function_exists( 'register_block_type' ) ) {
-			register_block_type(
-				self::BLOCK_NAME,
-				[
-					'attributes'      => [
-						'category'   => [
-							'type' => 'string',
-						],
-						'className'  => [
-							'type' => 'string',
-						],
-						'textSource' => [
-							'type' => 'string',
-						],
-					],
-					'render_callback' => [ $this, 'render_block' ],
-				]
-			);
-		}
+		add_filter( 'render_block', [ $this, 'render_serialized_block' ], 10, 2 );
 	}
 
 	/**
-	 * Gets the markup of the dynamic 'AR Viewer' block, including its scripts.
+	 * For a certain block namespace, renders the block only as a comment.
 	 *
-	 * @param array $attributes The block attributes.
-	 * @return string The markup of the block.
-	 */
-	public function render_block( $attributes ) {
-		self::$instance_id++;
-		$post                 = get_post();
-		$attributes['postId'] = $post->ID;
-
-		$this->plugin->components->Asset->enqueue_style( Asset::STYLE_SLUG );
-
-		// The script that loads the Headless Blocks model isn't needed in the block editor.
-		if ( ! is_admin() ) {
-			$this->plugin->components->Asset->enqueue_script( Asset::FRONT_END_SCRIPT_SLUG );
-
-			wp_add_inline_script(
-				$this->plugin->components->Asset->get_prefixed_slug( Asset::FRONT_END_SCRIPT_SLUG ),
-				$this->get_inline_script( $attributes ),
-				'before'
-			);
-		}
-
-		ob_start();
-		?>
-		<div class="headless-blocks-block" data-block-instance="<?php echo esc_attr( self::$instance_id ); ?>"></div>
-		<?php
-
-		return ob_get_clean();
-	}
-
-	/**
-	 * Gets the inline script for the block attributes.
+	 * This allows parsing the block on the front-end,
+	 * and rendering it with the props.
+	 * There will be no block markup without additional rendering on the front-end.
 	 *
-	 * @param array $props The props to add to the variable.
-	 * @return string The inline script.
+	 * @param string $block_content The initial block content to filter.
+	 * @param array  $block         The block data.
+	 * @return string The block content.
 	 */
-	private function get_inline_script( $props ) {
-		ob_start();
-		?>
-		if ( 'undefined' === typeof HeadlessBlocksProps ) {
-			var HeadlessBlocksProps = {};
+	public function render_serialized_block( $block_content, $block ) {
```

Comment:
This could be moved into Genesis Custom Blocks, and could run when `apply_filters( 'genesis_custom_blocks_is_headless_mode' )`